pkg/serviceaccount/jwt.go

```diff
@@ -225,42 +225,38 @@ func (j *jwtTokenGenerator) GenerateToken(claims *jwt.Claims, privateClaims inte
 // JWTTokenAuthenticator authenticates tokens as JWT tokens produced by JWTTokenGenerator
 // Token signatures are verified using each of the given public keys until one works (allowing key rotation)
 // If lookup is true, the service account and secret referenced as claims inside the token are retrieved and verified with the provided ServiceAccountTokenGetter
-func JWTTokenAuthenticator(issuers []string, keys []interface{}, implicitAuds authenticator.Audiences, validator Validator) authenticator.Token {
+func JWTTokenAuthenticator[Claims any](issuers []string, keys []interface{}, implicitAuds authenticator.Audiences, validator Validator[Claims]) authenticator.Token {
```

Comment:
nit: name this PrivateClaims (applies everywhere)

Reply:
Done.